nautobot/extras/signals.py

```diff
@@ -286,6 +287,53 @@ def dynamic_group_membership_created(sender, instance, raw=False, **kwargs):
 pre_save.connect(dynamic_group_membership_created, sender=DynamicGroupMembership)
 
 
+def dynamic_group_eligible_groups_changed(sender, instance, **kwargs):
+    """
+    When a DynamicGroup is created or deleted, refresh the cache of eligible groups for the associated ContentType.
+
+    Can't change content_type_id on an existing instance, so no need to check for that.
+    """
+    if kwargs.get("created", None) is False:  # Using the same handler for post_save and post_delete
```

Comment:
Is there a specific need to only explicitly check for `False` or is evaluating falsiness good enough?

```suggestion
    if not kwargs.get("created", False):  # Using the same handler for post_save and post_delete
```

Reply:
Yes, because "created" is not provided in the `post_delete` signal but is in the `post_save` case. Therefore the scenarios we have is `(True, False, Unset)` and we only want to return in the `False` case.